instrumentation/rocketmq-client-4.8/testing/src/main/java/base/IntegrationTestBase.java

```diff
@@ -0,0 +1,158 @@
+    NamesrvController namesrvController =
+        new NamesrvController(namesrvConfig, nameServerNettyServerConfig);
+    try {
+      Assert.assertTrue(namesrvController.initialize());
+      logger.info("Name Server Start:{}", nameServerNettyServerConfig.getListenPort());
+      namesrvController.start();
+    } catch (Exception e) {
+      logger.info("Name Server start failed", e);
+      System.exit(1);
+    }
+    NAMESRV_CONTROLLERS.add(namesrvController);
+    return namesrvController;
+  }
+
+  public static BrokerController createAndStartBroker(String nsAddr) {
+    String baseDir = createBaseDir();
+    BrokerConfig brokerConfig = new BrokerConfig();
+    MessageStoreConfig storeConfig = new MessageStoreConfig();
+    brokerConfig.setBrokerName(BROKER_NAME_PREFIX + BROKER_INDEX.getAndIncrement());
+    brokerConfig.setBrokerIP1("127.0.0.1");
+    brokerConfig.setNamesrvAddr(nsAddr);
+    brokerConfig.setEnablePropertyFilter(true);
+    storeConfig.setStorePathRootDir(baseDir);
+    storeConfig.setStorePathCommitLog(baseDir + SEP + "commitlog");
+    storeConfig.setMappedFileSizeCommitLog(COMMIT_LOG_SIZE);
+    storeConfig.setMaxIndexNum(INDEX_NUM);
+    storeConfig.setMaxHashSlotNum(INDEX_NUM * 4);
+    return createAndStartBroker(storeConfig, brokerConfig);
+  }
+
+  public static BrokerController createAndStartBroker(
+      MessageStoreConfig storeConfig, BrokerConfig brokerConfig) {
+    NettyServerConfig nettyServerConfig = new NettyServerConfig();
+    NettyClientConfig nettyClientConfig = new NettyClientConfig();
+    nettyServerConfig.setListenPort(nextPort());
+    storeConfig.setHaListenPort(nextPort());
+    BrokerController brokerController =
+        new BrokerController(brokerConfig, nettyServerConfig, nettyClientConfig, storeConfig);
+    try {
+      Assert.assertTrue(brokerController.initialize());
+      logger.info(
+          "Broker Start name:{} addr:{}",
+          brokerConfig.getBrokerName(),
+          brokerController.getBrokerAddr());
+      brokerController.start();
+    } catch (Throwable t) {
+      logger.error("Broker start failed, will exit", t);
+      System.exit(1);
+    }
+    BROKER_CONTROLLERS.add(brokerController);
+    return brokerController;
+  }
+
+  public static boolean initTopic(
+      String topic, String nsAddr, String clusterName, int queueNumbers) {
+    long startTime = System.currentTimeMillis();
+    boolean createResult;
+
+    while (true) {
+      createResult = MQAdmin.createTopic(nsAddr, clusterName, topic, queueNumbers);
+      if (createResult) {
+        break;
+      } else if (System.currentTimeMillis() - startTime > topicCreateTime) {
+        Assert.fail(
+            String.format(
+                "topic[%s] is created failed after:%d ms",
+                topic, System.currentTimeMillis() - startTime));
+        break;
+      } else {
+        TestUtils.waitForMoment(500);
+        continue;
+      }
+    }
+
+    return createResult;
```

Comment:
`createResult` will always be true, in any other case an `AssertionError` is thrown